bitcoin/src/blockdata/weight.rs

```diff
@@ -132,28 +208,31 @@ mod tests {
 
     #[test]
     fn kilo_weight_constructor() {
-        assert_eq!(Weight(1_000), Weight::from_kwu(1).expect("expected weight unit"));
+        assert_eq!(
+            Weight(1_000),
+            Weight::from_kwu(1).expect("weight unit from kwu does not overflow")
+        );
```

Comment:
This change is objectively worse, 4 lines of code and the message is no more meaningful. The main purpose, IMO, to use `expect` in unit tests is to differentiate potentially multiple `unwrap`s (not the case here, but in general). The actual string is inconsequential and using a long one harms readability. This is an example of one of `rustfmt`s problems, but lets not discuss that again :)

Reply:
I think `unwrap` would be completely fine here. The line number is shown correctly in non-ancient rustc.

Reply:
I guess I don't really agree that is objectively worse.  Although, I can understand just moving to using unwraps in general instead of expect messages.  However having to consider when to use expect and when to use unwrap in a test is probably going to lead to inconsistencies.

Reply:
I tend to use `unwrap` when that part is not the point of the test. In this case the point is to check equality, it just needs to produce the value from a constant.

Reply:
Ok I've replaced `expect()` with `unwrap()` in the weight tests.